digdag-core/src/main/java/io/digdag/core/database/BasicDatabaseStoreManager.java

```diff
@@ -380,62 +455,78 @@ private void validateTransactionAndCommit(Handle handle)
 
     public <T> T autoCommit(AutoCommitAction<T, D> action)
     {
-        try {
-            return transactionRetryExecutor.runInterruptible(() -> {
-                try (Handle handle = dbi.open()) {
-                    return action.call(handle, handle.attach(daoIface));
-                }
-            });
-        }
-        catch (RetryGiveupException ex) {
-            Throwable innerException = ex.getCause();
-            throw Throwables.propagate(innerException);
+        Handle nested = currentTransaction.get();
+        if (nested != null) {
+            return action.call(nested, nested.attach(daoIface));
         }
-        catch (InterruptedException ex) {
-            throw Throwables.propagate(ex);
+        else {
+            try {
+                return transactionRetryExecutor.runInterruptible(() -> {
+                    try (Handle handle = dbi.open()) {
```

Comment:
Is it possible to call a nested transactional methods like `transaction()` in `action.call()`? If so, this `handle` is needed to be set to `currentTransaction`?